src/mlio/CMakeLists.txt

```diff
@@ -113,7 +113,7 @@ endif()
 
 target_link_libraries(mlio
     PRIVATE
-        absl::strings dlpack::dlpack fmt::fmt protobuf::libprotobuf TBB::tbb
+        absl::strings dlpack::dlpack fmt::fmt protobuf::libprotobuf TBB::tbb ${OpenCV_LIBS}
```

Comment:
Avoid using the old way of declaring dependencies via variables. In modern CMake you use targets instead. I just checked the CMake config file of OpenCV. The ones you need are `opencv_core`, `opencv_imgproc`, `opencv_imgcodecs`. See the file `/usr/local/lib/cmake/opencv4/OpenCVModules.cmake` on Mac.

Reply:
got it, thanks for the suggestion

Reply:
is there a way to pass the `MLIO_BUILD_IMAGE_READER` parameter to this file as well? the build fails if `MLIO_BUILD_IMAGE_READER` is set to `OFF`.

Reply:
See [this](https://github.com/awslabs/ml-io/blob/master/src/mlio/CMakeLists.txt#L107) and [this](https://github.com/awslabs/ml-io/blob/master/src/mlio/CMakeLists.txt#L79) for two different approaches.

The first option simply defines a macro that you can check in your C++ file. The second option ignores the C++ file all together. My suggestion is to use the second option for all internal implementation files and use the first option for the file that has your `image_reader` type. If it is enabled compile the actual `image_reader`, if not, have a "placeholder" implementation of `image_reader` that just throws a `not_supported_error` in all of its public methods.

A third option is to use CMake's [`configure_file`](https://cmake.org/cmake/help/v3.2/command/configure_file.html) command, but the first two options should work.